test/e2e/service_latency.go

```diff
@@ -144,58 +165,155 @@ func runServiceLatencies(f *Framework, inParallel, total int) (output []time.Dur
 	return output, nil
 }
 
-func singleServiceLatency(f *Framework, i int) (time.Duration, error) {
-	// Make an RC with a single pod.
-	cfg := RCConfig{
-		Client:       f.Client,
-		Image:        "gcr.io/google_containers/pause:1.0",
-		Name:         fmt.Sprintf("trial-%v", i),
-		Namespace:    f.Namespace.Name,
-		Replicas:     1,
-		PollInterval: time.Second,
+type endpointQuery struct {
```

Comment:
As it stands at the moment, the second for loop will silently deadlock if total is not equal to the sum of errs and duration.  Wouldn't it be better to put a basic timeout on these reads so that you could produce a decent failure message, rather than a silent deadlock.
e.g here's a simple implementation:
http://blog.golang.org/go-concurrency-patterns-timing-out-and